dvc/ignore.py

```diff
@@ -244,7 +261,7 @@ def _update_sub_repo(self, path, ignore_trie: PathStringTrie):
             return
 
         root, dname = os.path.split(path)
-        pattern_info = PatternInfo(f"/{dname}/", f"in sub_repo:{dname}")
+        pattern_info = PatternInfo(f"/{dname}/", SubrepoException(dname))
```

Comment:
This is quite odd to include an exception like that. Maybe there is some nicer way to handle this case? Like, do we even subrepo trie anymore?

From the outside perspective (e.g. from `dvc/command/check_ignore.py`) we could very quickly and easilly check if path is a subrepo, by using something like:

```
repo = Repo(dirpath) # will walk up the tree from dirpath in search of .dvc dir
if self.repo.root_dir != repo.root_dir:
    # we are in subrepo
```

unless we have a nicer way of doing it. E.g. maybe have a `SubrepoPatternInfo`(or some flag in PatterInfo) so that in you could either raise in `_ignore_details` like you do right now or, maybe, just check for that on CLI level.

Reply:
@efiop , I raise Exception here is because the return value of `git check-ignore`. In Git:
- return 0 when the file is matched by some pattern (both include or ignored )
- return 1 when the file is not matched by any pattern.
- return 128 when the file is out of repo ( both in parent path or in a subrepo )

But the 0 and 1 here are counterintuitive, many had asked why not set 0 for missed or matches by include pattern and 1 only for ignored one.

Reply:
@karajan1001 I mean that it is odd to include an actual exception instance as a `file_info` in `PatternInfo`.